python/src/scipp/compat/mantid.py

```diff
@@ -82,25 +82,49 @@ def make_detector_info(ws):
                                         labels={'spectrum': spectrum}))
 
 
+def md_dimension(mantid_dim, index):
+    patterns = ["^q.*{0}$".format(coord) for coord in ['x', 'y', 'z']]
+    q_dims = [sc.Dim.Qx, sc.Dim.Qy, sc.Dim.Qz]
+    pattern_result = zip(patterns, q_dims)
+    if mantid_dim.getMDFrame().isQ():
+        for pattern, result in pattern_result:
+            if re.search(pattern, mantid_dim.name, re.IGNORECASE):
+                return result
+    return [sc.Dim.X, sc.Dim.Y, sc.Dim.Z][index]
+
+
+def md_unit(frame):
+    known_md_units = {
+        "Angstrom^-1": sc.units.dimensionless / sc.units.angstrom,
+        "r.l.u": sc.units.dimensionless
```

Comment:
This seems to imply that the X/Y/Z returned above is not actually real space, but just some dummy dimension label?

Reply:
If I can reasonably establish what the units are from the input frame, I can further this.